homeassistant/components/madvr/config_flow.py

```diff
@@ -0,0 +1,153 @@
+"""Config flow for the integration."""
+
+import asyncio
+import logging
+
+import aiohttp
+from madvr.madvr import HeartBeatError, Madvr
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigEntry, ConfigFlowResult, OptionsFlow
+from homeassistant.core import callback
+
+from .const import DOMAIN
+from .coordinator import MadVRCoordinator
+from .utils import CannotConnect, cancel_tasks
+from .wakeonlan import send_magic_packet
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class MadVRConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for the integration."""
+
+    async def async_step_user(self, user_input=None) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors = {}
+
+        if user_input is not None:
+            host = user_input["host"]
+            port = user_input.get("port", 44077)
+            mac = user_input["mac"]
+            keep_power_on = user_input["keep_power_on"]
+            try:
+                await self._test_connection(host, port, mac, keep_power_on)
+                return self.async_create_entry(
+                    title=user_input["name"], data=user_input
+                )
+            except CannotConnect:
+                _LOGGER.error("CannotConnect error caught")
+                errors["base"] = "cannot_connect"
+                self.context["user_input"] = user_input
+                # allow user to skip connection test
+                return await self.async_step_confirm()
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required("name"): str,
+                    vol.Required("host"): str,
+                    vol.Required("mac"): str,
+                    vol.Optional("port", default=44077): int,
+                    vol.Optional("keep_power_on", default=False): bool,
```

Comment:
Why does `__init__.py` take into account that these values are optional as they always get a value from the looks of it